pulser/sequence.py

```diff
@@ -12,32 +12,60 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from __future__ import annotations
+
 from collections import namedtuple
-from collections.abc import Iterable
+from collections.abc import Callable, Generator, Iterable, Set
 import copy
 from functools import wraps
 from itertools import chain
 import json
+from typing import Any, cast, NamedTuple, Optional, Tuple, Union
 import warnings
 
-import numpy as np
-
-import pulser
-from pulser.pulse import Pulse
-from pulser.devices import MockDevice
-from pulser.devices._device_datacls import Device
-from pulser.json.coders import PulserEncoder, PulserDecoder
-from pulser.json.utils import obj_to_dict
+from pulser.register import Register
 from pulser.parametrized import Parametrized, Variable
+from pulser.json.utils import obj_to_dict
+from pulser.json.coders import PulserEncoder, PulserDecoder
+from pulser.devices._device_datacls import Device
+from pulser.devices import MockDevice
+from pulser.pulse import Pulse
+from pulser.channels import Channel
 from pulser._seq_drawer import draw_sequence
+import pulser
+from numpy.typing import ArrayLike
+import numpy as np
```

Comment:
Please put these back on the second block of imports. For future reference, the PEP 8 style convention is to divide the imports into three blocks: 1. built-ins, 2. third-party libraries, 3. Local application/library specific imports.

Also, I think `import pulser` looks better at the top of the third block, but this is just a nit.